### PR TITLE
Cylinder-plane contact class and tests

### PR DESCRIPTION
**Pull request by Ali-7800**
7th contact module PR mentioned in [Issue](https://github.com/GazzolaLab/PyElastica/issues/113) 
This PR includes:

- Added `CylinderPlaneContact` class.
- Refactored `apply_normal_force_numba_rigid_body` to `_calculate_contact_forces_cylinder_plane.`
- Added tests for `CylinderPlaneContact` and fixed some typos.

### CODE REVIEW

**Review by armantekinalp — APPROVED** [review]
LGTM

File: tests/test_interaction.py [hunk 1]
```diff
@@ -1010,3 +1011,42 @@ def test_slender_body_matrix_product_only_xy(self, n_elem):
         slender_body_theory.apply_forces(rod)
 
         assert_allclose(correct_forces, rod.external_forces, atol=Tolerance.atol())
+
```
Comment (skim0119) [inline tests/test_interaction.py:1014]: Lets remove this block. We shouldn't have test case checking deprecation error/warning message. I'll add a commit to ignore.

[b1fa5d4](https://github.com/Ali-7800/PyElastica/commit/b1fa5d406198115df09f84c5d5b50547b59a1b22)

**Review by skim0119 — APPROVED** [review]
Minor change

**Review by ilianasiriziba — APPROVED** [review]
LGTM

**Pull request merged**